src/main/java/appeng/core/api/client/ApiCellModelRegistry.java

```diff
@@ -0,0 +1,106 @@
+    private static final ResourceLocation MODEL_CELL_ITEMS_4K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/4k_item_cell");
+    private static final ResourceLocation MODEL_CELL_ITEMS_16K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/16k_item_cell");
+    private static final ResourceLocation MODEL_CELL_ITEMS_64K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/64k_item_cell");
+    private static final ResourceLocation MODEL_CELL_FLUIDS_1K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/1k_fluid_cell");
+    private static final ResourceLocation MODEL_CELL_FLUIDS_4K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/4k_fluid_cell");
+    private static final ResourceLocation MODEL_CELL_FLUIDS_16K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/16k_fluid_cell");
+    private static final ResourceLocation MODEL_CELL_FLUIDS_64K = new ResourceLocation(
+            "appliedenergistics2:block/drive/cells/64k_fluid_cell");
+
+    private static final ResourceLocation[] MODELS = { MODEL_BASE, MODEL_CELL_EMPTY, MODEL_CELL_ITEMS_1K,
+            MODEL_CELL_ITEMS_4K, MODEL_CELL_ITEMS_16K, MODEL_CELL_ITEMS_64K, MODEL_CELL_FLUIDS_1K, MODEL_CELL_FLUIDS_4K,
+            MODEL_CELL_FLUIDS_16K, MODEL_CELL_FLUIDS_64K };
+
+    public static void registerModels() {
+        Arrays.stream(MODELS).forEach(ModelLoader::addSpecialModel);
+    }
+
+    private final Map<Item, ResourceLocation> registry;
+
+    public ApiCellModelRegistry(ApiDefinitions definitions) {
+        this.registry = new IdentityHashMap<>();
+        this.registry.put(definitions.items().cell1k().item(), MODEL_CELL_ITEMS_1K);
+        this.registry.put(definitions.items().cell4k().item(), MODEL_CELL_ITEMS_4K);
+        this.registry.put(definitions.items().cell16k().item(), MODEL_CELL_ITEMS_16K);
+        this.registry.put(definitions.items().cell64k().item(), MODEL_CELL_ITEMS_64K);
+        this.registry.put(definitions.items().fluidCell1k().item(), MODEL_CELL_FLUIDS_1K);
+        this.registry.put(definitions.items().fluidCell4k().item(), MODEL_CELL_FLUIDS_4K);
+        this.registry.put(definitions.items().fluidCell16k().item(), MODEL_CELL_FLUIDS_16K);
+        this.registry.put(definitions.items().fluidCell64k().item(), MODEL_CELL_FLUIDS_64K);
+    }
+
+    @Override
+    public boolean registerModel(Item item, ResourceLocation model) {
+        Preconditions.checkNotNull(item);
+        Preconditions.checkNotNull(model);
+        Preconditions.checkArgument(!this.registry.containsKey(item), "Cannot register an item twice.");
+
+        return this.registry.put(item, model) != null;
+    }
+
+    @Override
+    @Nonnull
+    public ResourceLocation model(@Nonnull Item item) {
+        Preconditions.checkNotNull(item);
+
+        return this.registry.get(item);
```

Comment:
This violates the annotation. Should return a default model (any, really), if none is registered.

Reply:
Would certainly make sense here.

Reply:
Correction: Currently it's even unused. Mostly in case some addon might want to lookup existing ones. So might not be ideal to return a default one, which is shared with another one like a 1k cell. Otherwise there is no way to tell, if there is a custom model or not.